lib/handler/redirect-handler.js

```diff
@@ -227,9 +227,10 @@ function cleanRequestHeaders (headers, removeContent, unknownOrigin) {
       }
     }
   } else if (headers && typeof headers === 'object') {
-    for (const key of Object.keys(headers)) {
+    const entries = headers instanceof Headers ? headers.entries() : Object.entries(headers)
```

Comment:
this won't work for Maps, only Headers

Reply:
maybe better to check for `Symbol.Iterable`

Reply:
@iiAku can you create a follow up PR?

```js
const entries = utils.isIterable(headers) ? headers : Object.entries(headers)
```

Reply:
this also only works with the global Headers, not Headers imported from undici

Reply:
> @iiAku can you create a follow up PR?
> 
> ```js
> const entries = utils.isIterable(headers) ? headers : Object.entries(headers)
> ```

Yes I can look into this, thanks for reviewing raising that point @KhafraDev